File: src/api/private/keys.ts
import { SessionHandle } from "~/models";
import forge from "node-forge";

type Keys = {
  iv: forge.util.ByteBuffer;
  key: forge.util.ByteBuffer;
};

export const aesKeys = (session: SessionHandle, forceEmptyIV = false): Keys => {
  const iv = forge.util.createBuffer(forceEmptyIV ? "" : session.information.aesIV);
  const key = forge.util.createBuffer(session.information.aesKey);

  return { iv, key };
};

File: library/src/commonMain/kotlin/api/private/keys.kt
package api.private

import io.ktor.utils.io.core.*
import models.SessionHandle
import models.SessionInformation

data class Keys(
    val iv: ByteArray,
    val key: ByteArray
)

@OptIn(ExperimentalStdlibApi::class)
fun aesKeys (sessionInfo: SessionInformation, forceEmptyIV: Boolean = false): Keys {
    val iv = (if (forceEmptyIV) "" else sessionInfo.aesIV).hexToByteArray()
    val key = sessionInfo.aesKey.hexToByteArray()

    return Keys(iv = iv, key = key)
}